modules/openapi-generator/src/main/resources/Java/modelEnum.mustache

```diff
@@ -56,7 +56,7 @@ import com.google.gson.stream.JsonWriter;
 {{/jackson}}
   public static {{{datatypeWithEnum}}}{{^datatypeWithEnum}}{{{classname}}}{{/datatypeWithEnum}} fromValue({{{dataType}}} value) {
     for ({{{datatypeWithEnum}}}{{^datatypeWithEnum}}{{{classname}}}{{/datatypeWithEnum}} b : {{{datatypeWithEnum}}}{{^datatypeWithEnum}}{{{classname}}}{{/datatypeWithEnum}}.values()) {
-      if (b.value.equals(value)) {
+      if (b.value.{{^isString}}equals{{/isString}}{{#isString}}equalsIgnoreCase{{/isString}}(value)) {
```

Comment:
as for me you should compare enum's constant name. so should be something like this

`if (b.value.equals(value){{#isString}} || b.name().equals(value){{/isString}}) {`